### ISSUE TITLE
PDBIO.set_structure() KeyError when residue/atom not in chain A

### ISSUE THREAD

**Issue by platinhom** [biopython/biopython#1912]
### The Case

When use PDBIO to write a residue/atom which is not in chain A, it will get `KeyError`. 

In the `set_structure` method, it will set the self.structure Chain ID to object Chain ID('B'), the next step to add the residue in the chain, the chain A is not exist now.

```python
## Source codes for PDBIO.set_structure
def set_structure(self, pdb_object):
      ..................
                  # When object is Residue
                   sb.init_chain('A')
                    if pdb_object.level == "R":
                        try:
                            parent_id = pdb_object.parent.id
                            sb.structure[0]['A'].id = parent_id
                        except Exception:
                            pass
                        sb.structure[0]['A'].add(pdb_object.copy())
```

I think to change `sb.structure[0]['A'].add(pdb_object.copy())` to `sb.structure[0][parent_id].add(pdb_object.copy())` will be better.

### The version information:

```
3.6.5 |Anaconda, Inc.| (default, Apr 26 2018, 08:42:37)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
CPython
Darwin-15.6.0-x86_64-i386-64bit
1.73
```

### Steps to reproduce

```python
import Bio
from Bio import PDB
from Bio.PDB import PDBIO, PDBParser, PDBList

pdbl = PDBList(pdb='.')
fp_3d91 = pdbl.retrieve_pdb_file('3d91', file_format='pdb', pdir='.')
p = PDBParser()
s = p.get_structure('3d91', 'pdb3d91.ent')
io = PDBIO()
## The following will sucess
io.set_structure(s[0]['B'])

## When use residue or atom, it will fail
io.set_structure(s[0]['B][10])
io.set_structure(s[0]['B'][10]['CA'])

## But chain A case will sucess
io.set_structure(s[0]['A'][10]['CA'])
```

**Comment by platinhom** [comment]
> From the code, there looks to be an undocumented assumption of model 0, chain A, when not setting an entire structure.
> 
> [biopython/Bio/PDB/PDBIO.py](https://github.com/biopython/biopython/blob/5ee5e69e649dbe17baefe3919e56e60b54f8e08f/Bio/PDB/PDBIO.py#L116)
> 
> Line 116 in [5ee5e69](/biopython/biopython/commit/5ee5e69e649dbe17baefe3919e56e60b54f8e08f)
> 
>  def set_structure(self, pdb_object): 
> Would you expect this to have used model 0, chain B in your example?
> 
> This is a guess, but you may need to first relabel your structure accordingly before using `PDBIO.set_structure(...)`?

Because I want to write out a protein pocket near a ligand. This pocket indeed contains two chains. I got error when writing chain B.

**Comment by peterjc** [comment]
This isn't going to address the limitation you've hit, but rather is thoughts on work-arounds or other ways of looking at it.

Can you not provide the full structure instead?

See also the ``Select`` class examples for how Thomas intended this to be used with ``PDBIO`` to export only a subset of a PDB file:

http://biopython.org/DIST/docs/tutorial/Tutorial.html

Or the ``Dice`` class if you know the residue values on the two chains?

**Comment by JoaoRodrigues** [comment]
This was addressed in #3006
